processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -808,6 +810,55 @@ public Object get()
                     }
 
                     @Override
+                    public NumericColumnSelector makeMathExpressionSelector(String expression)
+                    {
+                      final Expr parsed = Parser.parse(expression);
+                      final Set<String> required = Sets.newHashSet(Parser.findRequiredBindings(parsed));
+
+                      final Map<String, Supplier<Number>> values = Maps.newHashMapWithExpectedSize(required.size());
+                      for (String columnName : index.getColumnNames()) {
+                        if (!required.contains(columnName)) {
+                          continue;
+                        }
+                        final GenericColumn column = index.getColumn(columnName).getGenericColumn();
+                        if (column == null) {
+                          continue;
+                        }
+                        if (column.getType() == ValueType.FLOAT) {
+                          values.put(
+                              columnName, new Supplier<Number>()
+                              {
+                                @Override
+                                public Number get()
+                                {
+                                  return column.getFloatSingleValueRow(cursorOffset.getOffset());
+                                }
+                              }
+                          );
+                        } else if (column.getType() == ValueType.LONG) {
+                          values.put(
+                              columnName, new Supplier<Number>()
+                              {
+                                @Override
+                                public Number get()
+                                {
+                                  return column.getLongSingleValueRow(cursorOffset.getOffset());
+                                }
+                              }
+                          );
```

Comment:
Yep, there was double type here and looks like I removed them altogether by mistake. This would be rewritten with #2836, as commented above.